### PR TITLE
Enable CI builds on macOS and windows matching jack2 releases

### PR DESCRIPTION
**Pull request by falkTX**
Might take a few attempts, do not merge just yet.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -76,3 +81,157 @@ jobs:
         run: meson build && ninja -C build
       - name: Install jack-example-tools
         run: ninja -C build install
+  build_pawpaw_macos_intel:
```
Comment (dvzrv) [inline .github/workflows/build.yml:84]: Why do we need the macOS builds though?

Comment (falkTX) [reply .github/workflows/build.yml:84]: to test building against the official jack2 binaries.
the way homebrew builds things does not match jack2 releases. for one they do not support universal binaries.

**Comment by dvzrv** [comment]
AFAICT, the changes look good. However, they should be squashed and if possible split up in code change, build system change and CI change commits (as that is easier to bisect, review and also to cherry-pick by downstreams).

**Comment by falkTX** [comment]
Merged in main branch in 3 separate commits

**Pull request closed without merging**